pkg/client/client.go

```diff
@@ -71,10 +71,27 @@ func callTokenEndpoint(ctx context.Context, request any, authFn any, caller Toke
 	if err != nil {
 		return nil, err
 	}
-	tokenRes := new(oidc.AccessTokenResponse)
-	if err := httphelper.HttpRequest(caller.HttpClient(), req, &tokenRes); err != nil {
+
+	httpResp, err := caller.HttpClient().Do(req)
+	if err != nil {
 		return nil, err
 	}
+	defer httpResp.Body.Close()
+
+	resp := new(struct {
+		*oidc.AccessTokenResponse
+		*oidc.Error
+	})
+	if err = json.NewDecoder(httpResp.Body).Decode(resp); err != nil {
+		return nil, err
+	}
+
+	if httpResp.StatusCode < http.StatusOK || httpResp.StatusCode >= http.StatusBadRequest {
+		return nil, resp.Error
```

Comment:
If for some reason there was an error code and the OP did not return a proper JSON formatted OIDC error, there is a change that resp.Error will remain `nil` here.